### PR TITLE
Add recursive CTE in BQ to not supported list

### PR DESCRIPTION
**Pull request by Fleid**
## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/static/_redirects`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages

### CODE REVIEW

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ At present, model contracts are supported for:
 Model contracts are _not_ supported for:
 - Python models.
 - `ephemeral`-materialized SQL models.
+- Models with recursive CTEs in BigQuery
```
Comment (mirnawong1) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
- Models with recursive <Term id="cte" />'s in BigQuery
```

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 2]
```diff
@@ -38,6 +38,7 @@ At present, model contracts are supported for:
 Model contracts are _not_ supported for:
 - Python models.
 - `ephemeral`-materialized SQL models.
+- Models with recursive <Term id="cte" />'s in BigQuery
```
Comment (mirnawong1) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
- Models with recursive <Term id="cte" />'s in BigQuery.
```

**Comment by mirnawong1** [comment]
hey @Fleid  this looks great! i added a term id definition so users understood what cte's meant : https://docs-getdbt-aj4hfzz0z-dbt-labs.vercel.app/docs/collaborate/govern/model-contracts#where-are-contracts-supported
Screenshot 2024-02-26 at 10 43 17

looks great and merging! let me know if you have any strong opinions and thank you for opening this up!

**Review by mirnawong1 — APPROVED** [review]
lgtm!

**Comment by Fleid** [comment]
Thanks @mirnawong1, you're magic! :)

**Pull request merged**